src/games/strategy/triplea/ui/MapRouteDrawer.java

```diff
@@ -229,17 +229,18 @@ private void drawLineWithTranslate(Graphics2D graphics, Line2D line, double xOff
    * @return The "real" Point
    */
   private static Point getPointOnMap(Point point, int xOffset, int yOffset, int width, int height) {
-    double x = point.getX();
-    double y = point.getY();
+    int x = (int) point.getX();
+    int y = (int) point.getY();
     if (x - width > xOffset) {
-      point.setLocation(x - width, y);
+      point = new Point(x - width, y);
```

Comment:
hmm, actually you're not changing the parameter value - but you're changing the parameter reference. That is also  mutating the parameter. Same point applies, treat the params as final, in-only